src/uu/hashsum/src/hashsum.rs

```diff
@@ -276,11 +276,7 @@ pub fn uumain(mut args: impl uucore::Args) -> UResult<()> {
     };
     let check = matches.get_flag("check");
     let tag = matches.get_flag("tag");
-    let nonames = if binary_name == "b3sum" {
-        matches.get_flag("no-names")
-    } else {
-        false
-    };
+    let nonames = *matches.try_get_one("no-names").unwrap_or(Some(&false)).unwrap();
```

Comment:
While I change my code, I found another weird behavior of clap. 
`get_one("no-names")` panics inside the function call if the flag not exist (in test_check_sha1 test), so that it can't handle `unwrap_or()`.

So I used `try_get_one()` and it returns `Result<Option<T>, MatchesError>`. That's why I have two unwrap here..

Reply:
Oh yeah it should have been `try_get_one`! I made a mistake indeed :)